R/count_missed_doses.R

```diff
@@ -85,10 +77,80 @@ s_count_missed_doses <- function(x,
 #' * `a_count_missed_doses()` returns the corresponding list with formatted [rtables::CellValue()].
 #'
 #' @keywords internal
-a_count_missed_doses <- make_afun(
-  s_count_missed_doses,
-  .formats = c(n = "xx", count_fraction = format_count_fraction)
-)
+a_count_missed_doses <- function(x,
+                                 ...,
+                                 .stats = NULL,
+                                 .stat_names = NULL,
+                                 .formats = NULL,
+                                 .labels = NULL,
+                                 .indent_mods = NULL) {
+  dots_extra_args <- list(...)
+
+  # Check if there are user-defined functions
+  default_and_custom_stats_list <- .split_std_from_custom_stats(.stats)
+  .stats <- default_and_custom_stats_list$all_stats
+  custom_stat_functions <- default_and_custom_stats_list$custom_stats
+
+  # Adding automatically extra parameters to the statistic function (see ?rtables::additional_fun_params)
+  extra_afun_params <- retrieve_extra_afun_params(
+    names(dots_extra_args$.additional_fun_parameters)
+  )
+  dots_extra_args$.additional_fun_parameters <- NULL # After extraction we do not need them anymore
+
+  # Main statistical functions application
+  x_stats <- .apply_stat_functions(
+    default_stat_fnc = s_count_missed_doses,
+    custom_stat_fnc_list = custom_stat_functions,
+    args_list = c(
+      x = list(x),
+      extra_afun_params,
+      dots_extra_args
+    )
+  )
+
+  # Fill in with stats defaults if needed
+  .stats <- get_stats("count_missed_doses",
+    stats_in = .stats,
+    custom_stats_in = names(custom_stat_functions)
+  )
+
+  x_stats <- x_stats[.stats]
+  levels_per_stats <- lapply(x_stats, names)
+
+  # Fill in formats/indents/labels with custom input and defaults
+  .formats <- get_formats_from_stats(.stats, .formats, levels_per_stats)
+  .indent_mods <- get_indents_from_stats(.stats, .indent_mods, levels_per_stats)
+  if (is.null(.labels)) {
+    .labels <- sapply(.unlist_keep_nulls(x_stats), attr, "label")
+    .labels <- .labels[nzchar(.labels) & !sapply(.labels, is.null) & !is.na(.labels)]
+  }
+  .labels <- get_labels_from_stats(.stats, .labels, levels_per_stats)
```

Comment:
Could use these as defaults so that users can specify some labels but not all and still use the defaults (from the label attribute) for the rest.
```suggestion
  labels_def <- c(lapply(x_stats[["count_fraction"]], attr, "label"), tern_default_labels)
  .labels <- get_labels_from_stats(.stats, .labels, levels_per_stats, labels_def)
```

Reply:
In this way there are repetitions in labels_def right?

Reply:
There shouldn't be since we don't have defaults for variable levels - it will just add the level-specific labels (each corresponding to a number of missed doses) to the defaults so that the new ones are added but the "n" default label is kept